src/lib/components/payload-input-with-encoding.svelte

```diff
@@ -0,0 +1,61 @@
+<script context="module" lang="ts">
+  const encoding = ['json/plain', 'json/protobuf'] as const;
+  export type PayloadInputEncoding = (typeof encoding)[number];
+  export const isPayloadInputEncodingType = (
+    x: any, // eslint-disable-line @typescript-eslint/no-explicit-any
+  ): x is PayloadInputEncoding => encoding.includes(x);
+</script>
+
+<script lang="ts">
+  import { type Writable } from 'svelte/store';
+
+  import { onDestroy } from 'svelte';
+
+  import Card from '$lib/holocene/card.svelte';
+  import RadioGroup from '$lib/holocene/radio-input/radio-group.svelte';
+  import RadioInput from '$lib/holocene/radio-input/radio-input.svelte';
+  import { translate } from '$lib/i18n/translate';
+
+  import PayloadInput from './payload-input.svelte';
+
+  export let input: string;
+  export let encoding: Writable<PayloadInputEncoding>;
+  export let error = false;
+  export let resetValues = false;
+  export let loading = false;
+
+  $: {
+    if (resetValues) {
+      clearValues();
+    }
+  }
+
+  const clearValues = () => {
+    $encoding = 'json/plain';
+  };
+
+  onDestroy(() => {
```

Comment:
This is obviously _not_ a change you need to make, but this could probably just be `onDestroy(clearValues)`.